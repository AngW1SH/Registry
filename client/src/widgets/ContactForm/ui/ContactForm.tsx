import { SendContactFormEmail } from "@/features/SendContactFormEmail";
import { FC } from "react";

interface ContactFormProps {}

const ContactForm: FC<ContactFormProps> = () => {
  return (
    <div className="flex gap-32">
      <div>
        <h2 className="text-4xl font-medium">Заказчикам</h2>
        <div className="pt-8" />
        <p className="text-[0.9375rem] font-medium leading-6">
          Если у Вас есть запрос на
          <br />
          сотрудничество и создание проекта,
          <br />
          заполните онлайн-заявку.
          <br />
          Наш представитель свяжется с вами
          <br />в ближайшее время
        </p>
      </div>
      <SendContactFormEmail className="w-80" />
    </div>
  );
};

export default ContactForm;
