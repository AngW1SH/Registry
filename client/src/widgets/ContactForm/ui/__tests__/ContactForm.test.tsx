import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ContactForm from "../ContactForm";

describe("ContactForm Widget UI", () => {
  it("renders the component title", () => {
    const { getByText } = render(<ContactForm />);

    expect(getByText(/заказчикам/i)).toBeInTheDocument();
  });

  it("renders name and email input fields", () => {
    const { getByPlaceholderText } = render(<ContactForm />);

    const name = getByPlaceholderText(/имя/i);
    const email = getByPlaceholderText(/e-mail/i);

    expect(name).toBeInTheDocument();
    expect(email).toBeInTheDocument();
  });

  test("snapshot test", () => {
    const { asFragment } = render(<ContactForm />);

    expect(asFragment()).toMatchSnapshot();
  });
});
