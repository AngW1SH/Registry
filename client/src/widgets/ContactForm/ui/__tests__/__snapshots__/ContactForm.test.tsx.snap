// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Widget UI snapshot test 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-10 sm:flex-row md:gap-32"
  >
    <div>
      <h2
        class="text-4xl font-medium"
      >
        Заказчикам
      </h2>
      <div
        class="pt-4 sm:pt-8"
      />
      <p
        class="text-[0.9375rem] font-medium leading-6 [&>br]:hidden sm:[&>br]:block"
      >
        Если у Вас есть запрос на
        <br />
        сотрудничество и создание проекта,
        <br />
        заполните онлайн-заявку.
        <br />
        Наш представитель свяжется с вами
        <br />
        в ближайшее время
      </p>
    </div>
    <form
      class="flex flex-col w-full sm:w-80"
    >
      <div
        class="flex flex-col "
      >
        <input
          class="border-b border-[#898989] bg-transparent pb-2 font-normal placeholder-#898989 outline-none"
          placeholder="Имя"
          value=""
        />
      </div>
      <div
        class="pt-5"
      />
      <div
        class="flex flex-col "
      >
        <input
          class="border-b border-[#898989] bg-transparent pb-2 font-normal placeholder-#898989 outline-none"
          placeholder="E-mail"
          value=""
        />
      </div>
      <div
        class="pt-5"
      />
      <button
        class="cursor-pointer rounded-3xl bg-primary p-3 font-bold text-white mt-auto w-max px-12"
        type="submit"
      >
        Отправить
      </button>
    </form>
  </div>
</DocumentFragment>
`;
